src/shared/tpm2-util.c

```diff
@@ -1087,6 +1089,16 @@ static int tpm2_get_srk_template(Tpm2Context *c, TPMI_ALG_PUBLIC alg, TPMT_PUBLI
         return log_debug_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Unsupported SRK alg: 0x%x.", alg);
 }
 
+/* Get the best supported SRK template. ECC is preferred, then RSA. */
+static int tpm2_get_best_srk_template(Tpm2Context *c, TPMT_PUBLIC *ret_template) {
+        if (tpm2_get_srk_template(c, TPM2_ALG_ECC, ret_template) == 0 ||
+            tpm2_get_srk_template(c, TPM2_ALG_RSA, ret_template) == 0)
```

Comment:
doesn't matter, but our usual rule is that < 0 is error and >= is some form of success, and hence we'd usually check success with >= 0 rather than == 0, even if we know this function will only ever return 0, never above 0,

Reply:
ack, updated